contracts/libraries/RMM01Lib.sol

```diff
@@ -100,19 +100,25 @@ library RMM01Lib {
             ? self.params.priorityFee
             : self.params.fee;
 
-        (data.virtualX, data.virtualY) =
-            self.getVirtualPoolReservesPerLiquidityInWad();
+        data.liquidity = self.liquidity;
+        (data.virtualX, data.virtualY) = self.getVirtualReservesWad();
         tau = self.computeTau(timestamp);
-        data.prevInvariant = invariantOf({
-            self: self,
-            R_x: data.virtualX,
-            R_y: data.virtualY,
-            timeRemainingSec: tau
-        });
+
+        uint256 R_x;
+        uint256 R_y;
+        if (sellAsset) {
+            R_x = data.virtualX.divWadDown(data.liquidity);
+            R_y = data.virtualY.divWadUp(data.liquidity);
+        } else {
+            R_x = data.virtualX.divWadUp(data.liquidity);
+            R_y = data.virtualY.divWadDown(data.liquidity);
+        }
```

Comment:
Scales the reserve per liquidity amounts advantageously for the contract, to match the actual swap logic.